samza-core/src/main/java/org/apache/samza/storage/TransactionalStateTaskRestoreManager.java

```diff
@@ -243,30 +243,19 @@ static StoreActions getStoreActions(
           timeSinceLastCheckpointInMs = System.currentTimeMillis() -
               checkpointedChangelogOffset.getCheckpointId().getMillis();
         }
-      
-        // if the clean.store.start config is set, delete the currentDir, restore from oldest offset to checkpointed
-        if (storageEngine.getStoreProperties().isPersistedToDisk() && new StorageConfig(
-          config).getCleanLoggedStoreDirsOnStart(storeName)) {
-          File currentDir = storageManagerUtil.getTaskStoreDir(nonLoggedStoreBaseDirectory, storeName, taskName, taskMode);
-          LOG.info("Marking current directory: {} for store: {} in task: {}.", currentDir, storeName, taskName);
-          storeDirsToDelete.put(storeName, currentDir);
-          LOG.info("Marking restore offsets for store: {} in task: {} to {}, {} ", storeName, taskName, oldestOffset, checkpointedOffset);
-          storesToRestore.put(storeName, new RestoreOffsets(oldestOffset, checkpointedOffset));
-          return;
-        }
 
         // if the clean.store.start config is set, delete the currentDir, restore from oldest offset to checkpointed
         if (storageEngine.getStoreProperties().isPersistedToDisk() && new StorageConfig(
```

Comment:
I'd suggest you also check if the store is logged store explicitly. It is possible logged store directory and non-logged store directory happens to be the same directory and we will by-mistake clear the stores that are non-logged.

Reply:
good catch.

@rmatharu was your intention in PR https://github.com/apache/samza/pull/1200 for both logged and non-logged stores to be clear based on this config?

Reply:
to recap: @mynameborat 's question was with regard to side inputs, which are persisted and non-logged but are not to be deleted. this is a non-issue since side inputs are not hooked in through TransactionalStateTaskRestoreManager but instead TaskSideInputStorageManager. @mynameborat feel to comment if i am misunderstanding our discussion.